astropy/io/fits/_tiled_compression/tiled_compression.py

```diff
@@ -253,11 +254,13 @@ def _get_compression_setting(header, name, default):
 
 def _column_dtype(compressed_coldefs, column_name):
     tform = compressed_coldefs[column_name].format
-    if tform[2] == "B":
+    if tform.startswith("1"):
+        tform = tform[1:]
+    if tform[1] == "B":
         dtype = np.uint8
-    elif tform[2] == "I":
+    elif tform[1] == "I":
         dtype = ">i2"
-    elif tform[2] == "J":
+    elif tform[1] == "J":
         dtype = ">i4"
```

Comment:
Actually this if/else block can probably be replaced by `tform.recformat.dtype`.

Reply:
I just tried this but got some test failures, so I might leave this to another follow-up PR as I need time to investigate what is going wrong.

Reply:
Sure, it's just that it seems we could reuse more code from `io.fits.column` for the header parsing and maybe for the verification, but fine to keep this for later.